### DISCUSSION TITLE
Image for alpine and debian/ubuntu

### DISCUSSION THREAD

**Discussion by Dexus** [Secure-Compliance-Solutions-LLC/GVM-Docker#260] [General]
Hi @pixelsquared @austinsonger,

lately I've noticed that Greenbone products only really run well in their "environments", i.e. on Debian variants.

But since I can't work on the problems full time, we should consider going back to Debian/Ubuntu and build alpine on the side.

As it turns out in the last days, alpine and also various other Linux variants are not really supported and it doesn't seem like it's really wanted.

So my question is do we want to support both Linux variants? Or rather limit it to one?

I would like to continue to work on Alpine without various segfaults and co as a system to use, but that will not be possible in a short time.

Let's discuss!

Greetings, Josef

**Comment by cfi-gb** [comment]
> and it doesn't seem like it's really wanted.

@bjoernricks might be able to say more but AFAIC it's not about "wanted" but about "can't". The reference system for GVM is "Debian Buster".

Support for issues / problems originating from other systems / environments (like musl c library on Alpine) is not possible from Greenbone side due to various reasons and such support (like via Github PR to add musl compatibility) needs to be provided by the community via e.g. Github PR. AFAICT such PRs have been merged already in various cases with more or less delay.

**Reply by Dexus** [reply]
Hi @cfi-gb 
I understand that, it may have been worded a bit wrong and too pointed. If I am resented for that, I apologize.

Of course, such PR were included, but even there so far when I look through the history, only waved through, there is the possibility to test on all distributions, for example, there is the github action/workflows - just as an idea. 

We will surely get it done, but the question remains how and with what effort.

Unfortunately, many problems only become apparent later, with segfaults etc..

**Reply by austinsonger** [reply]
I think the best route at this point would be the shortest route to image that isn't getting segfaults. but it looks like segfault is happening on OpenVAS docker image? If we are not getting segfault on GVM docker image, I wouldn't see the need to change both. But it would be best to make a decision from the community.

**Reply by Dexus** [reply]
So far I see the segfaults only on gvm-docker with openvas included. Don't have so much OpenVAS docker images in use currently.

Seen binaries that are segfaults:
- gvmd
- openvas
- nmap

**Reply by austinsonger** [reply]
Whatever is best for the community, then is probably the route we should go.

**Reply by bjoernricks** [reply]
To be clear for the moment we as Greenbone support only our reference system which is currently Debian Buster. That doesn't mean we don't accept patches for all our components to fix issues on other systems from the community. We are a free software company and the power of free software is about contributions and sharing. It just means we don't maintain such patches. We can't ensure that code for the other systems always works and if it creates unnecessary maintenance burden for other code we may remove it.
